src/ingest/SpecRepo.cpp

```diff
@@ -154,17 +157,22 @@ void SpecRepo::genPatternSpec(const nebula::meta::PatternMacro macro,
   // moving step based on macro granularity
   const auto step = Macro::seconds(macro);
 
+  // fill in custom macros
+  auto enumeratedPathTemplates = Macro::enumeratePathsWithCustomMacros(pathTemplate, table->macroValues);
+
   // from now going back step by step until exceeding maxSeconds
   size_t count = 0;
   while (count < maxSeconds) {
     const auto watermark = now - count;
-    // populate the file paths for given time point
-    const auto path = Macro::materialize(macro, pathTemplate, watermark);
+    for (const auto& pathWithoutTime : enumeratedPathTemplates) {
+      // populate the file paths for given time point
+      const auto path = Macro::materialize(macro, pathWithoutTime, watermark);
 
-    // list files in the path and generate spec perf file from it
-    genSpecPerFile(table, version, fs->list(path), specs, watermark);
+      // list files in the path and generate spec perf file from it
+      genSpecPerFile(table, version, fs->list(path), specs, watermark);
 
-    count += step;
+      count += step;
```

Comment:
seems like a bug if nesting the time increment per enumerated path. 

what do you think about moving the for loop (line 167) to the outer of the while loop (at line 162)? in that case, we repeat/restart the time range for every enumerated path, logic seems clear enough.